include/sys/zfs_stat.h

```diff
@@ -44,11 +45,21 @@ typedef struct zfs_stat {
 	uint64_t	zs_mode;
 	uint64_t	zs_links;
 	uint64_t	zs_ctime[2];
+	uint64_t	zs_parent;
+	uint64_t	zs_flags;
```

Comment:
I'll looks like this would mainly impact zc_zoneid, which is only used by jail/unjail, which only seems to be in freebsd. I could return the stat data (including zs_parent & zs_flags) in an nvlist and keep the size of zfs_cmt_t the same.